### PR TITLE
Create Console class

### PR DESCRIPTION
**Pull request by sarahabimay**
- create Console class
- Introduce BoardDisplayer 
- Handle invalid move exceptions

@maikon
@jsuchy

### CODE REVIEW

File: lib/human_player.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def get_opponent
     Mark::O if @mark == Mark::X 
   end
 
-  def get_next_move(board)
-    @display.ask_player_for_move(board)
+  def get_next_move
+    @display.ask_player_for_move(@mark)
```
Comment (gemcfadyen) [inline lib/human_player.rb:17]: I think you can refer to the member variables without the '@' sign in the class they are defined (except in the constructor)

File: lib/player_factory.rb [hunk 2]
```diff
@@ -1,14 +1,18 @@
+require "human_player"
+
 class PlayerFactory
   attr_reader :players
```
Comment (gemcfadyen) [inline lib/player_factory.rb:5]: You may want to put the attr_readers in a private section, as they are public here.

File: lib/human_player.rb (hunk 1 above)
Comment (Maikon) [inline lib/human_player.rb:17]: Yep, you can only do that if you have `attr_reader` or `attr_accessor` which essentially create getter/setter methods with that name.

File: spec/console_spec.rb [hunk 3]
```diff
@@ -0,0 +1,104 @@
+require "console"
+require "stringio"
+
+RSpec.describe Console do
+  let(:input) { StringIO.new }
```
Comment (jsuchy) [inline spec/console_spec.rb:5]: All of this setup smells. My general rule is no more than two (maybe three) collaborators for any given class, I count at least 8 here.

**Pull request merged**